doxygen/contributor_help_pages/require_meta.md

```diff
@@ -0,0 +1,191 @@
+
+For overloading classes and structs with this scheme we create an initial forward definition with a `void` non-type template parameter. Then the class overloads use the requires in place of the non-type template parameter.
+
+```cpp
+template <typename T, typename = void>
+class a_class;
+
+/**
+ * Overload for standard vectors with \ref stan::scalar_type
+ *  of \ref stan::math::var
+ */
+template <typename T>
+class a_class<T, require_std_vector_st<is_var, T>> {
+  // ...
+};
+```
+
+As noted in the docs above, `a_class` has an overload specifically for standard vectors with @ref stan::scalar_type 's of @ref stan::math::var .
+
+Below covers the general themes for all of the [`requires`](@ref require_meta) found
+in the Stan math library. Any `*` should be thought of as a wildcard where a type
+traits name is put in its place.
+
+- `requires_*_t`: A template parameter `T` must satisfy the require in order for
+the overload to be available.
+
+```cpp
+ // Works for stan::math::var types
+ template <typename T, require_var_t<T>* = nullptr>
+ auto add(T&& x, T&& y) { return x + y; }
+```
+
+- `require_not_*_t` : A template parameter `T` must *not* satisfy the require in order for
+the overload to be availabe.
+
+*NOTE:* The `not` version of requires should be used sparingly. Often `requires` are used to specify, conceptually, what types a function should accept. Defining a function by the types it cannot accept can make understanding what goes into a function more difficult and error prone.
+
+```cpp
+ // Works for non standard vector types
+ template <typename T, require_not_std_vector_t<T>* = nullptr>
+ auto add(T&& x, T&& y) { return x + y; }
+```
+
+- `require_all_*_t` : Takes a parameter pack of types to enable if all types
+satisfy the check.
+
+```cpp
+ // Works if T1 and T2 are complex
+ template <typename T1, typename T2,
+   require_all_complex_t<T1, T2>* = nullptr>
+ auto add(T1&& x, T2&& y) { return x + y; }
+```
+
+- `require_any_*_t` : Takes a parameter pack of types to enable if any of
+the types satisfy the check.
+
+```cpp
+ // Works if either T1 or T2 enherit from EigenBase
+ template <typename T1, typename T2, require_any_eigen_t<T1, T2>* = nullptr>
+ auto add(T1&& x, T2&& y) { return x + y; }
+```
+
```

Comment:
This is a super helpful section, by the way!

Reply:
Ty!